### PR TITLE
refactor(heatmap): simplify get maximum text length logic

### PR DESCRIPTION
**Pull request by wsuwt**
## Description

Simplify get maximum text length logic.

## Type of change

- [x] Refactor

## Checklist
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] I have checked my code and corrected any misspellings

### CODE REVIEW

File: packages/elements/src/heatmap/helpers/text.ts [hunk 1]
```diff
@@ -95,22 +95,12 @@ const getMaximumTextWidth = (ctx: CanvasRenderingContext2D, cells: HeatmapCell[]
...
-  for (let i = 0; i < labels.length; i++) {
-    if (maxLengthText.length >= labels[i].length) {
-      maxLengthText = maxLengthText;
-    }
-    else {
-      maxLengthText = labels[i];
-    }
-  }
-  return maxLengthText;
+  return labels.reduce((previousText, currentText) => previousText.length > currentText.length ? previousText : currentText);
```
Comment (dtanp-rft) [inline packages/elements/src/heatmap/helpers/text.ts]: I'd say rename var from `xxText` to `xxLabel` would be better.
And `@returns Text with maximum length` -> `@returns Longest label in the array`

Btw, method name could be better :)

Comment (wsuwt) [reply packages/elements/src/heatmap/helpers/text.ts]: Agreed and fixed.

File: packages/elements/src/heatmap/helpers/text.ts [hunk 2]
```diff
@@ -95,22 +95,12 @@ const getMaximumTextWidth = (ctx: CanvasRenderingContext2D, cells: HeatmapCell[]
...
-      maxLengthText = labels[i];
-    }
-  }
-  return maxLengthText;
+const getMaximumLabelWidth = (labels: string[]): string => {
+  return labels.reduce((previousLabel, currentLabel) => previousLabel.length > currentLabel.length ? previousLabel : currentLabel);
 };
 
-export { getResponsiveFontSize, getMaximumTextWidth, getMaximumLabelTextWidth, MIN_FONT_SIZE, MAX_FONT_SIZE };
+export { getResponsiveFontSize, getMaximumTextWidth, getMaximumLabelWidth, MIN_FONT_SIZE, MAX_FONT_SIZE };
```
Comment (dtanp-rft) [inline packages/elements/src/heatmap/helpers/text.ts]: What I actually don't like about the method name is that it ends with `Width` so I would assume that this method returns width but actually, it returns the longest label.

You can see that `getMaximumTextWidth` and `getMaximumLabelWidth` have similar name while the first one is returning width (which makes sense for its name).

Comment (wsuwt) [reply packages/elements/src/heatmap/helpers/text.ts]: That's true. I'll go for `getLongestLabel()` instead.

Comment (wsuwt) [reply packages/elements/src/heatmap/helpers/text.ts]: Actually we might not even need to extract this method at all. I can just place this line onto the code itself.

Comment (dtanp-rft) [reply packages/elements/src/heatmap/helpers/text.ts]: I like that :) I want to push one line comment and I will approve this PR

**Review by dtanp-rft — APPROVED** [review]

**Pull request merged**